packages/integration-karma/test/mixed-shadow-mode/synthetic-behavior/index.spec.js

```diff
@@ -0,0 +1,90 @@
+        beforeEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', true);
+        });
+        afterEach(() => {
+            setFeatureFlagForTest('ENABLE_MIXED_SHADOW_MODE', false);
+        });
+
+        const scenarios = [
+            {
+                Component: ParentAnyChildAny,
+                tagName: 'x-parent-any-child-any',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentAnyChildReset,
+                tagName: 'x-parent-any-child-reset',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentResetChildAny,
+                tagName: 'x-parent-reset-child-any',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentResetChildReset,
+                tagName: 'x-parent-reset-child-reset',
+                nativeLeaf: false,
+            },
+            {
+                Component: ParentLightChildAny,
+                tagName: 'x-parent-light-child-any',
+                nativeLeaf: true,
+            },
+            {
+                Component: ParentLightChildReset,
+                tagName: 'x-parent-light-child-reset',
+                nativeLeaf: false,
+            },
+        ];
+
+        scenarios.forEach(({ Component, nativeLeaf, tagName }) => {
+            describe(tagName, () => {
+                let elm;
+                let ids;
+
+                beforeEach(() => {
+                    elm = createElement(tagName, { is: Component });
+                    document.body.appendChild(elm);
+                    ids = extractDataIds(elm);
+                });
+
+                it('should render the shadow root in the correct shadow mode', () => {
+                    expect(isNativeShadowRootInstance(ids.leaf.shadowRoot)).toEqual(nativeLeaf);
+                });
+
+                it(`should set the scope attributes only in synthetic shadow`, () => {
+                    const attributes = Array.from(ids.div.attributes)
+                        .map((_) => _.name)
+                        .filter((_) => _ !== 'data-id');
+                    expect(attributes.length).toEqual(nativeLeaf ? 0 : 1);
+                    expect(getComputedStyle(ids.div).color).toEqual('rgb(0, 0, 255)');
+                });
+
+                it(`should mangle the IDs only in synthetic shadow`, () => {
+                    if (nativeLeaf) {
+                        expect(ids.label.id).toEqual('foo');
+                        expect(ids.input.getAttribute('aria-labelledby')).toEqual('foo');
+                    } else {
+                        expect(ids.label.id).not.toEqual('foo');
+                        expect(ids.input.getAttribute('aria-labelledby')).not.toEqual('foo');
+                        expect(ids.label.id).toEqual(ids.input.getAttribute('aria-labelledby'));
+                    }
+                });
+            });
+        });
+    });
```

Comment:
There are probably even more tests we can add here:

- grandparent with reset/any, parent with reset/any, leaf with reset/any (8 tests total)
- slotter with reset/any, slottable with reset/any, leaf with reset/any (8 tests total)

I got tired just writing the tests I did, but it's probably valuable to add these at some point.